src/groupeddataframe/groupeddataframe.jl

```diff
@@ -303,7 +303,7 @@ Base.getindex(gd::GroupedDataFrame, idx...) =
     getindex(gd, Base.to_indices(gd, idx)...)
 
 # The allowed key types for dictionary-like indexing
-const GroupKeyTypes = Union{GroupKey, Tuple, NamedTuple}
+const GroupKeyTypes = Union{GroupKey, Tuple, NamedTuple, Dict{Symbol}, Dict{<:AbstractString}}
```

Comment:
Maybe this should be `AbstractDict`? Same elsewhere.

Reply:
why not. This probably requires a careful consideration if this does not introduce problems though (probably not).

For sure we can provide a conversion from `GroupKey` to a dictionary only for a concrete type (`Dict`).